File: apps/mobile/lib/features/community/domain/models/rescue_application.dart
import 'package:meta/meta.dart';

enum RescueApplicationStatus { pending, approved, rejected, adminOverridden }

enum ApplicantRole { foster, adopter }

@immutable
class RescueApplication {
  final String id;
  final String rescuePostId;
  final int applicantId;
  final ApplicantRole applicantRole;
  final String? message;
  final String? experienceSummary;
  final bool hasVehicle;
  final RescueApplicationStatus status;
  final DateTime? posterReviewedAt;
  final DateTime? adminReviewedAt;
  final int? adminReviewerId;
  final DateTime createdAt;

  const RescueApplication({
    required this.id,
    required this.rescuePostId,
    required this.applicantId,
    required this.applicantRole,
    this.message,
    this.experienceSummary,
    required this.hasVehicle,
    required this.status,
    this.posterReviewedAt,
    this.adminReviewedAt,
    this.adminReviewerId,
    required this.createdAt,
  });
}

File: apps/api/HappyPaws.Domain/Entities/RescueApplication.cs
using System;
using HappyPaws.Domain.Enums;

namespace HappyPaws.Domain.Entities;

public class RescueApplication
{
    public Guid Id { get; set; }
    public Guid RescuePostId { get; set; }
    public long ApplicantId { get; set; }
    public ApplicantRole ApplicantRole { get; set; }
    public string? Message { get; set; }
    public string? ExperienceSummary { get; set; }
    public bool HasVehicle { get; set; } = false;
    public RescueApplicationStatus Status { get; set; } = RescueApplicationStatus.Pending;
    public DateTimeOffset? PosterReviewedAt { get; set; }
    public DateTimeOffset? AdminReviewedAt { get; set; }
    public long? AdminReviewerId { get; set; }
    public DateTimeOffset CreatedAt { get; set; }

    public Post RescuePost { get; set; } = null!;
    public User Applicant { get; set; } = null!;
    public User? AdminReviewer { get; set; }
}
